src/main/java/fi/okm/jod/yksilo/validation/FreeTextValidator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.validation;
+
+import fi.okm.jod.yksilo.domain.LocalizedString;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class FreeTextValidator implements ConstraintValidator<FreeText, LocalizedString> {
+
+  private static final Pattern PAT =
+      Pattern.compile("[\\w\\p{Punct}\\p{gc=S}\\x20\\n\\r]*", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
Or maybe `\\x20\\n\\r` -> `\\s` (white space in general, somewhat more permissive, allows also e.g. tabs).